src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/TagsController.cs

```diff
@@ -0,0 +1,60 @@
+﻿using System.Collections.Generic;
+using System.Globalization;
+using System.Web.Http;
+using Orchard.ContentManagement;
+using Orchard.Core.Title.Models;
+using Orchard.Localization;
+using Orchard.Logging;
+using Orchard.Taxonomies.Helpers;
+using Orchard.Taxonomies.Models;
+using Orchard.Taxonomies.Services;
+using System.Linq;
+
+namespace Orchard.Taxonomies.Controllers {
+    public class TagsController : ApiController {
+        private readonly ITaxonomyService _taxonomyService;
+        private readonly IContentManager _contentManager;
+        public Localizer T { get; set; }
+        protected ILogger Logger { get; set; }
+
+        public TagsController(
+            ITaxonomyService taxonomyService,
+            IContentManager contentManager) {
+            _taxonomyService = taxonomyService;
+            T = NullLocalizer.Instance;
+            _contentManager = contentManager;
+            Logger = NullLogger.Instance;
+        }
+
+        public IEnumerable<TagDto> Get(int taxonomyId, bool leavesOnly, string query) {
+            if (string.IsNullOrEmpty(query)) return new List<TagDto>();
```

Comment:
There should be some security check here, at least the same as what the content editor is requiring to list taxonomy terms. Which means at least Admin access, plus other taxonomy specific ones.